### DISCUSSION TITLE
Password verifying

### DISCUSSION THREAD

**Discussion by barbaracrlp** [filamentphp/filament#12543] [Help]
### Package

Panel builder

### Package Version

v3

### How can we help you?

I'm connecting to a DB with alreadey made users, in that table I used a different hashing method.How can I change Filament's password verification so it uses my way of hashing? I got my own 
method of verify but I don't know where to put it.
![image](https://github.com/filamentphp/filament/assets/121516133/055ed73f-4289-4d71-b90f-4796e4218c2e)

**Comment by leandrocfe** [comment]
- Create a custom login class extending the default Login
- Customize the authenticate method

```php
use Filament\Pages\Auth\Login as AuthLogin;
class Login extends AuthLogin
{
    public function authenticate(): ?LoginResponse
    {
         //...
    }
}
```

Add the custom class in the panel provider

```php
$panel
->login(Login::class)
...
```
